indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -575,58 +542,231 @@ public Response getCompleteTasks(
       @Context final HttpServletRequest req
   )
   {
-    final Function<String, Task> taskFunction = taskId -> {
-      final Optional<Task> optionalTask = taskStorageQueryAdapter.getTask(taskId);
-      if (!optionalTask.isPresent()) {
+    return getTasks("complete", null, null, maxTaskStatuses, null, req);
+  }
+
+  @GET
+  @Path("/tasks")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getTasks(
+      @QueryParam("state") final String state,
+      @QueryParam("datasource") final String dataSource,
+      @PathParam("interval") final String interval,
+      @QueryParam("max") final Integer maxCompletedTasks,
+      @QueryParam("type") final String type,
+      @Context final HttpServletRequest req
+  )
+  {
+
+    //check for valid state
+    if (state != null) {
+      List validStates = ImmutableList.of("pending", "waiting", "running", "complete");
```

Comment:
It'd be more hygienic if this were a constant, like

```java
public static final API_TASK_STATES = ImmutableList.of("pending", "waiting", "running", "complete")
```

Reply:
sure, done.